slides/linear_model.md

```diff
@@ -0,0 +1,260 @@
+class: titlepage
+
+.header[MOOC Machine learning with scikit-learn]
+
+# Linear Model
+
+This lesson covers the linear models. These are basic models, easy to understand and fast to train
+
+<img src="../scikit-learn-logo.svg">
+
+???
+
+Linear models are easy to understand and fast to train, 
+they give us fair baselines
+
+
+---
+# Outline
+
+* What is a linear model?
+* Linear model for regression & classification
+* How to avoid overfitting?
+
+
+---
+# Adult census
+
+.very-small[
+
+| Age | Workclass | Education    | Marital-status     | Occupation         | Relationship | Race  | Sex  | Capital-gain | Hours-per-week | Native-country | Salary |
+| --- | --------- | ------------ | ------------------ | ------------------ | ------------ | ----- | ---- | ------------ | -------------- | -------------- | ----- |
+| 25  | Private   | 11th         | Never-married      | Machine-op-inspct  | Own-child    | Black | Male | 0            | 40             | United-States  | $45k |
+| 38  | Private   | HS-grad      | Married-civ-spouse | Farming-fishing    | Husband     | White  | Male | 0            | 50             | United-States   | $40k |
+| 28  | Local-gov | Assoc-acdm   | Married-civ-spouse | Protective-serv    | Husband      | White | Male | 0            | 40             | United-States   | $60k  |
+| 44  | Private   | Some-college | Married-civ-spouse | Machine-op-inspct  | Husband      | Black | Male | 7688         | 40             | United-States   | $52k  |
+
+]
+Salary = *.4 x* education + *.2 x* hours-per-week + *.1 x* age + ...
+
+???
+Adult census is here a bit modify, instead we have the value of the salary for every one.
+So we can see it as a regression problem
+
+Salary could be a linear combination of the feature (explanatory variable). 
+
+
+
+---
+# Linear regression
+
+Predict the value of the target **y**  
+given some observation **X**
+
+.shift-down.pull-left.shift-left[<img src="../figures/linear_data.svg" width="100%">]
+
+???
+For illustration purpose, let's consider one dimensionnal observation, 
+e.g. salary should be explain by education level (nb of year of study)
+
+---
+# Linear regression
+A linear model is a slope "as close as possible" from datapoint
+The blue curve is the prediction for each **x**
+
+.shift-down.pull-left.shift-left[<img src="../figures/linear_fit.svg" width="100%">]
+.pull-right.very-small[```python
+from sklearn.linear_model import LinearRegression
```

Comment:
It would be super cool if this line fit entirely on the left part without having a slider.
I don't know if we should reduce the padding or change the way to import?

Reply:
Did it. 
But now I need to find a way to add space between the figure and the code.